packages/flutter/test/widgets/semantics_tester.dart

```diff
@@ -350,6 +354,14 @@ class SemanticsTester {
   /// tester.
   SemanticsTester(this.tester) {
     _semanticsHandle = tester.binding.pipelineOwner.ensureSemantics();
+
+    // This _extra_ clean-up is needed for the case when a test fails and
+    // therefore fails to call dispose() explicitly. The test is still required
+    // to call dispose() explicitly, because the semanticsOwner check is
+    // performed irrespective of whether the owner was created via
+    // SemanticsTester or directly. When the test succeeds, this tear-down
+    // becomes a no-op.
+    addTearDown(dispose);
```

Comment:
Good find!  I have been meaning to check on this because it causes extra failures when a test fails, making it harder to see the real failure sometimes.